### ISSUE TITLE
Stack element's placement and page number in TOC disagree when setting unbreakable:true.

### ISSUE THREAD

**Issue by bigmatik** [bpampuch/pdfmake#1288]
First of all, let me thank the authors for this great tool. I hope the issue found may help it improve.

If we want to tie some paragraphs/images together we can use a `stack` and set the property `unbreakable:true`. If ---at the same time--- we want to include an element in the stack in the TOC we may find that the actual page where the stack is placed and the page appearing in the TOC do not agree.
 
Here's how to reproduce it: depending on the preceding elements both the header and the image are placed in the next page, yet the page appearing in the TOC does not agree with the new placement.

In the code below set `numberOfEmptyLines` between `25` and `35` (at the given width the image seems to take up `10` lines of height) to observe the disagreement. Run it in the playground.

```javascript
var emptyLines = '';
var numberOfEmptyLines = 25;
for (var i=0; i<numberOfEmptyLines; i++)
{
    emptyLines = emptyLines + '\n';
}

var headAndImage = {   
    unbreakable:true,
    stack: [
        {
            text: "Image header",
            style: "subheader",
            tocItem: true
        },
        {
            image:'sampleImage.jpg', width:315
            
        }
    ]
};

var dd = {
  styles: {
    header: {
      fontSize: 18,
      bold: true
    },
    subheader: {
      fontSize: 15,
      bold: true
    }
  },
  content: [
    {
      toc: {
        title: {text: "INDEX", style: "header"}
      }
    },
    {
      text:'\n\n'
    },
    {
      text: "Ipsum header",
      style: "subheader",
      tocItem: true
    },
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin finibus non mi eget euismod. Cras sit amet lacus ipsum. Phasellus in elementum erat. Duis euismod quam id mauris pharetra volutpat eu quis libero. Aliquam et consequat risus. Donec velit ligula, molestie a lectus sit amet, fermentum maximus mi. Morbi egestas elit vitae metus ullamcorper, sit amet efficitur tellus viverra. Praesent sagittis leo ante, sed blandit mi tincidunt a. Sed ullamcorper est a mi mattis, ut commodo turpis sollicitudin. Sed odio nunc, rutrum vitae ante dapibus, posuere ultrices sem. Nulla facilisi. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Vivamus semper pulvinar lorem, sit amet ultricies nulla congue ornare. Etiam euismod, purus ac varius ullamcorper, ipsum tortor tempor est, volutpat congue dui odio at ex. Nullam fermentum nibh iaculis nisi efficitur interdum.\n\n',

    emptyLines, // the empty lines go between the paragraph and the header&Image pair.

    headAndImage // the header and the image go here.
  ]
};

```

Instead of within a `stack` you may group the desired elements in a `table` row together with `dontBreakRows:true`, but to no avail regarding the TOC.

```javascript
var headAndImage = {layout:'noBorders',
  table: { 
    dontBreakRows: true,
    body:[
      [ 
        [ 
          { 
              text: "Image header",
              style: "subheader",
              tocItem: true
          },
          { 
              image:'sampleImage.jpg', width:315
          } 
        ] 
      ] 
    ] 
  } 
};
```

**Comment by petermate** [comment]
I have the same problem and I was just wondering whether there is a solution or workaround for this issue?

**Comment by liborm85** [comment]
Fixed by commit https://github.com/bpampuch/pdfmake/commit/d3af785ac49cf841eb0bcb3e2369421ee55fef26 (in master) and https://github.com/bpampuch/pdfmake/commit/f096d16144548b6000ebb89bcb20733c65160067 (in 0.1 branch).

**Comment by liborm85** [comment]
Released in 0.1.54.
